lib/cache/repositories/impressions/redis_repository.rb

```diff
@@ -59,13 +59,17 @@ def clear
             end
 
             impressions
+          rescue StandardError => e
+            @config.logger.error("Exception while clearing impressions cache: #{e}")
           end
 
           private
 
           # Get all sets by prefix
           def impression_keys
             @adapter.find_sets_by_prefix("#{@config.redis_namespace}/*/impressions.*")
+          rescue StandardError => e
```

Comment:
@patricioe `impression_keys` will return `nil` and no iteration would happen, i.e. `clear` method will return `nil` as well